modules/apps/headless/headless-discovery/headless-discovery-impl/src/main/java/com/liferay/headless/discovery/internal/jaxrs/application/HeadlessDiscoveryOpenAPIApplication.java

```diff
@@ -73,6 +75,18 @@
 		RuntimeDTO runtimeDTO = _jaxrsServiceRuntime.getRuntimeDTO();
 
 		for (ApplicationDTO applicationDTO : runtimeDTO.applicationDTOs) {
+			String name = applicationDTO.name;
+
+			if (StringUtil.startsWith(name, "O_")) {
```

Comment:
:S

Any other way? Do we add ObjectDefinition or something more evident than this? This is coupling Discovery with Object in a string way that is not ideal but I don't think there is another easy way to do it but it's also not very obvious...

Reply:
I know... but the only thing I find to check which application we were processing was the applicationDTO.name. I'll give it a second thought to check if there is a better way to check it without coupling with implementation details

Reply:
I've tried to find a better solution but I can't find something easy to implement. Maybe we can use the base property instead of the name but doesn't seem it is much better than this.

I think it could be possible to create some mechanism relying on a service registry where we can register the applicationDTO that has restricted company access and which company they have access to. Then we can check the registry to filter in Headless Discovery. We should also need to filter the direct requests to this application with restricted company access when the company is different than the one selected. Seems to be a possible approach but I think it requires quite an effort.

Do you think it makes sense to make this effort now? Can we go with an easy solution and then implement a better one?

Reply:
I was thinking on relying on the resources that are generated by ourselves and we set the name (and are accesible by runtimeDTO) but it's pretty similar to the current solution.

Let's leave it like this to merge this sooner because it has been 2 weeks since they pinged us with the issue.